pkg/pool-weighted/test/WeightedPoolFactory.test.ts

```diff
@@ -26,17 +29,30 @@ describe('WeightedPoolFactory', function () {
   let createTime: BigNumber;
 
   sharedBeforeEach('deploy factory & tokens', async () => {
-    const vault = await Vault.create();
+    vault = await Vault.create();
 
     factory = await deploy('WeightedPoolFactory', { args: [vault.address] });
     createTime = await currentTimestamp();
 
     tokens = await TokenList.create(['MKR', 'DAI', 'SNX', 'BAT'], { sorted: true });
+    // Deploy Asset manager
+    assetManager = await deploy('v2-asset-manager-utils/TestAssetManager', {
```

Comment:
This is fine, but in the future you might want to consider just passing an EOA here to avoid coupling to this contract, and head scratching in the future trying to understand why this test fails if that contract later changes.